### PR TITLE
Replace http-client `got` with `ky`, improving browser compatibility

### LINKED ISSUE #655: Problem installing to npm project
Hello!
Please bear with me as I'm fairly new to npm.
I've tried incorporating this project into my npm project through "npm install musicbrainz-api" and followed the readme guide.  However, I ran into all kinds of errors - they all seem to relate to Webpack 5.

Some of the errors I'm getting:
BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

WARNING in ./node_modules/musicbrainz-api/lib/xml/xml-isrc.js
Module Warning (from ./node_modules/source-map-loader/dist/cjs.js):
Failed to parse source map from ...

Uncaught ReferenceError: process is not defined
    at ./node_modules/@szmarczak/http-timer/dist/source/index.js

So, at this point, I might resort to uninstalling it and referencing it for my own hand-crafted code - it would be easier that way!
Thanks

### PR DESCRIPTION
**Pull request by Haschikeks**
Replace http-client got with ky, improving browser compatibility

Closes #655

### CODE REVIEW

**Review by Borewit — APPROVED** [review]
LGTM

File: lib/musicbrainz-api.ts [hunk 1]
```diff
@@ -15,13 +15,11 @@ import { DigestAuth } from './digest-auth';
 import { RateLimiter } from './rate-limiter';
 import * as mb from './musicbrainz.types';
 
-import got, { Options } from 'got';
+import ky, {Options, ResponsePromise} from 'ky-universal';
```
Comment (Borewit) [inline lib/musicbrainz-api.ts:18]: Just keep the code styke
```suggestion
import ky, { Options, ResponsePromise } from 'ky-universal';
```

**Review by Borewit — CHANGES_REQUESTED** [review]

**Comment by Borewit** [comment]
Sorry, I did a wrong test, I thought all was working, but I forgot to recompile.

ky-universal is a pure ESM module since version [v0.9.0](https://github.com/sindresorhus/ky-universal/releases/tag/v0.9.0).
That is breaking the tests.

**Comment by Borewit** [comment]
When [downgrading to v0.8.2](https://github.com/Borewit/musicbrainz-api/tree/use-ky-v0.8.2) downgrading I run into errors in the ky module.

**Comment by Haschikeks** [comment]
> When [downgrading to v0.8.2](https://github.com/Borewit/musicbrainz-api/tree/use-ky-v0.8.2) downgrading I run into errors in the ky module.

This is because the [kv v0.26.0](https://github.com/sindresorhus/ky/releases/tag/v0.26.0) is a pure ESM module as well. Using v0.25.1 works on my end. It also runs the tests.

**Comment by Borewit** [comment]
Yet it not really a sustainable solution as these commonjs modules are no longer supported.

**Comment by Haschikeks** [comment]
I agree. I'll list possible solutions I see for this problem below.
For now, it would help to mention the fact, that the package is a node-only package. (Not everybody reads through open issues)

- Make `musicbrainz-api` an pure [ESM module](https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c), which would allow us to use the latest `ky` packages
- Search for other possible solutions to support browsers, that are not pure ESM packages
- make this package explicitly a node only package without support for running in a browser environment
- expose a fetcher api that the user must provide
  I've seen other repositories do this, I might have to search for an example
  - In the Browser this could be the native [`fetch`](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) API
  - In a node environment this could be any package the user likes to use for requests, e.g. [node-fetch](https://www.npmjs.com/package/node-fetch)

**Comment by Borewit** [comment]
All very good suggestions @Haschikeks.

> Make musicbrainz-api an pure [ESM module](https://gist.github.com/sindresorhus/a39789f98801d908bbc7ff3ecc99d99c), which would allow us to use the latest ky packages

Pure ESM is nice, but users will complain they cannot get it to work in their CommonJS projects. Yet we choose to do so.

> Search for other possible solutions to support browsers, that are not pure ESM packages.

Sure.

> Expose a fetcher api that the user must provide

I like that is well. At the same time, not all functions are unfortunately clean REST. Hence we need to work with cookies. This would make wiring a HTTP-client a little cumbersome and tricky.

I not tracking changes in MusicBrainz API at all, could be that they have better implementations nowadays.

> Make this package explicitly a node only package without support for running in a browser environment

Essentially close to the "do nothing" option, but document this restriction better. 

I have no strong opinion on the direction. If you want to do more work, and you favor any of the above, I will support you.
If you favor ESM, I can apply the same kind of changes as I have done in some other projects. We could also maintain ESM on a different branch, and possible release that as different NPM.

Are you avoiding the question to become a collaborator @Haschikeks?

**Comment by Borewit** [comment]
> But some time ago I accepted it. (At least thats what I think I did? Github shows me as a collaborator in this issue, so)

Ah my bad, for some reason I thought you did not.  So sorry.

> I can have a look at, how such an interface would need to look like and how we could handle the cookies. I'll also check the MusicBrainz-API in regards to the usage of cookies.

It are essentially the functions in which you see `formData`, which use the non-REST ways of publishing information, which require cookies and CSRF, as they follow the way normal web user would submit the data. Like the function `editEntity`, `addUrlToRecording`, `addIsrc`.

**Comment by Borewit** [comment]
@Haschikeks although there are a few challenges are remaining, a lot of the pre-requisites  have now been met. I like to make this module browser complaint. Possibly we move non compliant functions to a Node.js specific entry point) . Can you please consider to update and rebase this PR?

**Comment by Borewit** [comment]
@Haschikeks are you interested in continuing this work? Otherwise I might pick it up.

**Comment by Haschikeks** [comment]
Hi @Borewit,
I'm sorry for not replying.
Please go ahead and pick it up

**Comment by Borewit** [comment]
Closed in favor of #968, Ky was hardly adding any value. That it is throwing errors on some HTTP codes, made it only harder to work with (a HTTP redirect for instance).

**Pull request closed without merging**